Mailbox/Person.cs

```diff
@@ -1,4 +1,62 @@
-﻿namespace Mailbox
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Mailbox
 {
+    public struct Person : IEquatable<Person> 
+    {
+        private string FirstName { get; set; }
+        private string LastName { get; set; }
+
+        public Person(String firstName, String lastName)
+        {
+            if(firstName is null || lastName is null)
+            {
+                throw new ArgumentNullException();
+            }
+            
+            FirstName = firstName;
+            LastName = lastName;
+        }
+
+        public bool Equals([AllowNull]Person other)
+        {
+            if(LastName == other.LastName && FirstName == other.FirstName)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public override bool Equals(object? obj)
+        {
+            if (obj is null)
+            {
+                return false;
+            }
+            return Equals((Person)obj);
```

Comment:
This will throw a cast exception if a non-null object is passed into the equals method.
Such as:
```
Person p = new Person("Foo", "bar");
p.Equals(new object());
```

A better solution would be to use pattern matching.
```suggestion
            return obj is Person person && Equals(person);
```